dtcli/scripts/dt.py

```diff
@@ -1,821 +1,830 @@
+
+    try:
+        return dict(map(split_pair_and_verify_key, filter(None, re.split(r"(?<!\\)\/", value))))
+        return value
+    except ValueError:
+        raise click.BadParameter(f"format must be '/key0=value0/key1=value1/...' got: '{value}'")
+
+
+def edit_other_option_if_true(ctx, param, value, other_name, edit_callback):
+    if not value:
+        return
+    for p in ctx.command.params:
+        if isinstance(p, click.Option) and p.name == other_name:
+            edit_callback(p)
+
+
+def _genca(ca_cert_path, ca_key_path, force, subject, days_valid, ca_passphrase):
+    if force:
+        print("Forced generation option used. Already existing CA certificate files will be overwritten.")
+        utils.check_file_exists(ca_cert_path, utils.KeyGenerationError)
+        utils.check_file_exists(ca_key_path, utils.KeyGenerationError)
+        signing.generate_ca(
+            ca_cert_path,
+            ca_key_path,
+            subject,
+            datetime.datetime.today() + datetime.timedelta(days=days_valid),
+            ca_passphrase,
+        )
+        return
+
+    if utils.check_file_exists(ca_cert_path, utils.KeyGenerationError, warn_overwrite=False) and utils.check_file_exists(
+        ca_key_path, utils.KeyGenerationError, warn_overwrite=False
+    ):
+        raise utils.KeyGenerationError(
+            "CA certificate NOT generated! CA key and certificate already exist. Use --force option to generate anyway."
+        )
+
+    signing.generate_ca(
+        ca_cert_path,
+        ca_key_path,
+        subject,
+        datetime.datetime.today() + datetime.timedelta(days=days_valid),
+        ca_passphrase,
+    )
+
+
+def token_load(ctx, param, value):
+    """
+    Function load token to application.
+
+    First it checks if path is passed as argument.If not it takes
+    default token localization defined in constants.py as DEFAULT_TOKEN_PATH, else gets token from file
+    passed as argument. If file with token doesn't exist it checks if virtual variable DTCLI_API_TOKEN
+    exist and returns token if so or error if not.
+    """
+    try:
+        if value == '-':
+            value = const.DEFAULT_TOKEN_PATH
+
+        with open(value) as f:
+            try:
+                token = f.readlines()[0].rstrip()
+            except IndexError:
+                raise click.BadArgumentUsage("Token file exist but is empty. No token applied.")
+        return token
+    except FileNotFoundError:
+        token = os.getenv("DTCLI_API_TOKEN")
+        if token is None:
+            raise click.UsageError("Virtual environment DTCLI_API_TOKEN doesn't exist. No token applied.")
+        return token
+
+
+def parse_tenant_url(value: str) -> str:
+    if value.endswith("/"):
+        value = value[:-1]
+    pr = requests.models.PreparedRequest()
+    try:
+        pr.prepare_url(value, None)
+    except requests.exceptions.MissingSchema as e:
+        click.echo(f"{e}. Https:// will be added to URL and retried.", err=True)

```

Comment:
Not sure if we want to throw everything at the user as is and I'd love to see actual "warning" in the message